woodwork/statistics_utils/_get_dependence_dict.py

```diff
@@ -0,0 +1,201 @@
+import warnings
+from timeit import default_timer as timer
+
+import numpy as np
+import pandas as pd
+from sklearn.metrics.cluster import adjusted_mutual_info_score
+
+from ._bin_numeric_cols_into_categories import _bin_numeric_cols_into_categories
+
+from woodwork.accessor_utils import _is_dask_dataframe, _is_koalas_dataframe
+from woodwork.exceptions import SparseDataWarning
+from woodwork.utils import _update_progress, get_valid_mi_types
+
+
+def _get_dependence_dict(
+    dataframe,
+    measure,
+    num_bins=10,
+    nrows=None,
+    include_index=False,
+    callback=None,
+    extra_stats=False,
+    min_shared=25,
+    random_seed=0,
+):
+    """Calculates dependence measures between all pairs of columns in the DataFrame that
+    support measuring dependence. Logical Types that are supported are as
+    follows:  Boolean, Categorical, CountryCode, Datetime, Double, Integer, Ordinal,
```

Comment:
Wondering if we should make this supported types list more generic instead of listing specific types to make it less likely to get outdated as we change/add types? Are the nullable types supported? What about Age? If we just listed boolean, categorical, datetime and numeric columns instead, would that cover it?

Same comment applies to other docstrings where this is mentioned.